libgoal/libgoal.go

```diff
@@ -777,10 +777,10 @@ func (c *Client) ApplicationInformation(index uint64) (resp generatedV2.Applicat
 }
 
 // ApplicationBoxes takes an app's index and returns the names of boxes under it
-func (c *Client) ApplicationBoxes(index uint64) (resp generatedV2.BoxesResponse, err error) {
+func (c *Client) ApplicationBoxes(index uint64, boxNumMax *uint64) (resp generatedV2.BoxesResponse, err error) {
```

Comment:
nit: maybe make it a varargs param so you don't have to supply. Not sure what we usually do in this repo though.

Reply:
good point, I was hesitate about the design of passing in pointer, and golang does not allow for overloading function, or default argument.

So at this point, seemingly varargs fits better? I will change accordingly now.

Reply:
just changed as your suggestion